src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/InbreedingCoeff.java

```diff
@@ -0,0 +1,149 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.GenotypesContext;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.InfoFieldAnnotation;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Likelihood-based test for the inbreeding among samples
+ *
+ * <p>This annotation estimates whether there is evidence of inbreeding in a population. The higher the score, the higher the chance that there is inbreeding.</p>
+ *
+ * <h3>Statistical notes</h3>
+ * <p>The calculation is a continuous generalization of the Hardy-Weinberg test for disequilibrium that works well with limited coverage per sample. The output is a Phred-scaled p-value derived from running the HW test for disequilibrium with PL values. See the <a href="http://www.broadinstitute.org/gatk/guide/article?id=4732">method document on statistical tests</a> for a more detailed explanation of this statistical test.</p>
+ *
+ * <h3>Caveats</h3>
+ * <ul>
+ * <li>The Inbreeding Coefficient can only be calculated for cohorts containing at least 10 founder samples.</li>
+ * <li>This annotation is used in variant recalibration, but may not be appropriate for that purpose if the cohort being analyzed contains many closely related individuals.</li>
+ * <li>This annotation requires a valid pedigree file.</li>
+ * </ul>
+ *
+ */
+public final class InbreedingCoeff extends InfoFieldAnnotation {
+
+    private static final Logger logger = LogManager.getLogger(InbreedingCoeff.class);
+    private static final int MIN_SAMPLES = 10;
+    private final Set<String> founderIds;
+
+    public InbreedingCoeff(final Set<String> founderIds){
+        //If available, get the founder IDs and cache them. the IC will only be computed on founders then.
+        this.founderIds = founderIds;
+
+        if ( founderIds == null || founderIds.isEmpty() ) {
+            logger.warn("Annotation will not be calculated, must provide a valid PED file (-ped) from the command line.");
```

Comment:
This warning isn't accurate. The real issue is if there are fewer than 10 founders. Founders will be null if no pedigree is passed in. Instead, we should do something like the some of the other annotations where we can't output the warning until we get to the annotate() function. There we should count the genotypes and warn if that's < 10.